### PR TITLE
add TerrahubCommand::getDependecyStrategy (dependence option)

### LINKED ISSUE #719: Add new option for dependencies to cover related use cases in terrahub run action

### PR DESCRIPTION
**Pull request by const-andreyLu**
## Description

Fixes #719

## Type of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a change to the documentation

## How Has This Been Tested?

- [x] thub apply -I api -d ignore
- [x] thub apply -I api -d include

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have checked that my changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by const-andreyLu** [comment]
Suggest to add informative message in errors when dependency option can be used, something like 
```
To ignore or include dependencies use [--dependency][-d] option
```

**Comment by avozicov** [comment]
I like your suggestion @const-andreyLu .
Let's ask @eistrati and @euliancom to review your proposal ;)

File: src/terraform-command.js [hunk 1]
```diff
@@ -416,10 +445,9 @@ class TerraformCommand extends AbstractCommand {
 
     while (hashesToCheck.length) {
       const hash = hashesToCheck.pop();
-      const { dependsOn } = fullConfig[hash];
+      const node = fullConfig[hash];
 
-      dependsOn
-        .map(path => ConfigLoader.buildComponentHash(path))
+      Object.keys(node.dependsOn)
```
Comment (Leichtwind) [inline src/terraform-command.js]:
```suggestion
      Object.keys(dependsOn)
```

File: src/terraform-command.js [hunk 2]
```diff
@@ -416,10 +445,9 @@ class TerraformCommand extends AbstractCommand {
 
     while (hashesToCheck.length) {
       const hash = hashesToCheck.pop();
-      const { dependsOn } = fullConfig[hash];
+      const node = fullConfig[hash];
```
Comment (Leichtwind) [inline src/terraform-command.js]:
```suggestion
      const { dependsOn } = fullConfig[hash];
```

File: src/terraform-command.js [hunk 3]
```diff
@@ -149,15 +155,12 @@ class TerraformCommand extends AbstractCommand {
       exclude.length ? hash => !exclude.includes(config[hash].name) : null
     ].filter(Boolean);
 
-    Object.keys(config)
-      .filter(hash => filters.some(check => !check(hash)))
-      .forEach(hash => { delete config[hash]; });
 
     if (!Object.keys(config).length) {
```
Comment (Leichtwind) [inline src/terraform-command.js]: This check should be performed before returning the result

**Pull request merged**